backend.py

```diff
@@ -402,6 +402,13 @@ def add_repos(main_repositories, update_repositories, repos):
     if 'sources' in answers_pristine:
         for i in answers_pristine['sources']:
             repos = repository.repositoriesFromDefinition(i['media'], i['address'])
+            repo_gpgcheck = (answers.get('repo-gpgcheck', True) if i['repo_gpgcheck'] is None
+                             else i['repo_gpgcheck'])
+            gpgcheck = (answers.get('gpgcheck', True) if i['gpgcheck'] is None
+                        else i['gpgcheck'])
+            for repo in repos:
+                repo.setRepoGpgCheck(repo_gpgcheck)
+                repo.setGpgCheck(gpgcheck)
```

Comment:
This looks like a partial stepback from what was done in the first two commits. Would it make sense to change the first two commits direclty so that the third one doesn't have to move the code around?

Reply:
Yes it does.

Also, moving the code from a loop on `main_repositories` into a loop over all sources, as currently done, has the side effect of applying the `*gpgcheck` parameters to all sources, which is likely to cause issues when we suddenly perform the check on an update repo (since XenServer does not activate this check on those).